docs/02-guides/configuration.mdx

```diff
@@ -1,14 +1,394 @@
+
+:::note
+
+`slowValuesOnEachRequest` is a low level API in Front-Commerce. You shouldn't be
+need it in most cases.
+
+:::
+
+`slowValuesOnEachRequest` is a function that extracts configuration values from
+the current request. For instance, depending on the URL, the configuration
+`currentShopId` will get a different id and thus display different information.
+
+```js
+const slowValuesOnEachRequest = (req) => {
+  const url = req.originalUrl;
+  const shopId = getShopIdFromUrl(url);
+  return {
+    currentShopId: shopId,
+  };
+};
+```
+
+:::caution
+
+This part is named `slowValuesOnEachRequest` because we want to stress the fact
+that it can have a huge impact on your website performance. Avoid its usage as
+much as you can and try to use `values` instead. If this is the only way to
+setup your configuration make sure to memoize its result to limit the
+performance impact as much as possible.
+
+```js
+import memoize from "lodash/memoize";
+
+const getShopIdFromHostname = memoize((hostname) => {
+  /* The definition of the `currentShopId` variable should live here */
+  return {
+    currentShopId: currentShopId,
+  };
+});
+
+const slowValuesOnEachRequest = (req) => {
+  const hostname = req.hostname;
+  return getShopIdFromHostname(req.hostname);
+};
+```
+
+:::
+
+### `fetchRemoteConfiguration` <Label>(optional)</Label>
+
+:::note
+
+`fetchRemoteConfiguration` is a low level API in Front-Commerce. You shouldn't
+need it in most cases.
+
+:::
+
+`fetchRemoteConfiguration` is a function that is called at a later stage in
+Front-Commerce request handling cycle to receive a fully initialized `request`
+object, so that, for instance,
+[you can instantiate a loader to retrieve some configuration from Magento](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/commit/f57ccabae32ebe6243fc213485078cb1f98f8a30#9e3c152a167c50222b152728db1b5946d409ba89_0_29).
+
+:::caution
+
+Like `slowValuesOnEachRequest`, `fetchRemoteConfiguration` can have a huge
+impact on the performances. If you really need to implement it, please make sure
+to memoize its result to limit the performance impact as much as possible.
+
+:::
+
+## Inject a configuration provider
+
+Assuming that we have defined the following configuration provider:
+
+```ts title="extensions/custom-extension/configProvider.ts"
+export default {
+  name: "serviceProvider",
+  schema: () => ({
+    service: {
+      key: {
+        doc: "The key to get access to our remote service",
+        format: String,
+        default: null,
+        env: "FRONT_COMMERCE_SERVICE_KEY",
+      },
+      secret: {
+        doc: "The secret to get access to our remote service",
+        format: String,
+        default: null,
+        env: "FRONT_COMMERCE_SERVICE_SECRET",
+      },
+    },
+  }),
+};
+```
+
+You will then need to inject the config provider in your
+[extensions definition](/docs/3.x/api-reference/front-commerce-core/defineExtension).
```

Comment:
**issue:** I think the most common use case is to register it in the application

**suggestion:** split into 2 small sections "From your application configuration file" / "From an extension" illustrating both approaches.

Reply:
What is the "From your application configuration file"? the old method of using [register](http://localhost:3000/docs/2.x/advanced/server/configurations#register-a-configuration-provider)?